Rpc: introduce get_inflation_reward rpc call (#16278)
solana-labs/solana: docs/src/developing/clients/jsonrpc-api.md

{"jsonrpc":"2.0","result":{"epoch":100,"foundation":0.001,"total":0.149,"validator":0.148},"id":1}
```

### getInflationReward

> Returns the inflation reward for a list of addresses and an epoch

Reviewer: ```suggestion
Returns the inflation reward for a list of addresses for an epoch
```

#### Parameters:
- `<array>` - An array of addresses to query, as base-58 encoded strings

> - `<u64>` - (optional) An epoch for which the reward occurs. If omitted, the previous epoch will be used

Reviewer 2: Needs update, can refer to `getStakeActivation` for format

> - `<u64>` - (optional) An epoch for which the reward occurs (if omitted, the most recent epoch reward will be returned)

Reviewer: ```suggestion
- `<u64>` - (optional) An epoch for which the reward occurs.  If omitted, the previous epoch will be used
```

#### Results

The result field will be a JSON array with the following fields:

- `epoch: <u64>`, epoch
- `effective_slot: <u64>`, the slot in which the rewards are effective
- `amount: <u64>`, reward amount in lamports

> - `post_balance: <u64>`, post balance in lamports

Reviewer: ```suggestion
- `post_balance: <u64>`, post balance of the account in lamports
```
